hatchways/team-cosmic-crisp
# FS Dashboard

### What this PR does (required):
- Changed sign-up form to include first and last name - creates profile on user creation (removed username)
- Fetches all sitter profiles when user is not logged in. (isDogSitter marked to true and have price and city)
- When the user is logged in, fetches all sitter profiles excluding the current user
- You can click on a profile and route to profile details page
- Navbar now displays profile photo
- Added grow, fade animations and loading spinners to profile details and listing pages

Note: Currently, the search bar feature in the dashboard and the send request form in the profile details are not yet functional.
Also, ratings for sitter profiles have been hardcoded at the moment, I'm almost done working on the controllers to fetch reviews. Post that, ratings and reviews will be fetched from the API and only logged in users will be able to rate and create reviews.

### Screenshots / Videos (required):
### Sign Up Form
![1](https://user-images.githubusercontent.com/75184965/119676336-a2714680-bdf2-11eb-8d1c-ab1c396e8326.PNG)

### Public Dashboard
![2](https://user-images.githubusercontent.com/75184965/119676408-aef59f00-bdf2-11eb-81b0-6cfa86c304a5.PNG)

### Profile Details Page
![3](https://user-images.githubusercontent.com/75184965/119676492-bd43bb00-bdf2-11eb-8320-e3960791c7af.PNG)

### Private Dashboard
![4](https://user-images.githubusercontent.com/75184965/119676580-cdf43100-bdf2-11eb-99df-6c79f58a328a.PNG)


### Any information needed to test this feature (required):
- `npm start` on client | `npm run dev` on server
- Please make sure you are logged out
- Enter /listings in url to view all dog sitter profiles. Eg: `http://localhost:3000/listings`
- Click on listing to route to profile details
- Create an account or login using existing account, you will be taken to the listing page.

### Any issues with the current functionality (optional):
- None
